src/racer/ast.rs

```diff
@@ -841,6 +838,94 @@ impl<'c, 's, 'ast> visit::Visitor<'ast> for ExprTypeVisitor<'c, 's> {
                     }
                 }
             }
+            ExprKind::Binary(bin, ref left, ref right) => {
+                self.visit_expr(left);
+                let type_match = match self.result.take() {
+                    Some(Ty::Match(m)) => m,
+                    Some(Ty::PathSearch(ps)) => match ps.resolve_as_match(self.session) {
+                        Some(m) => m,
+                        _ => {
+                            return;
+                        }
+                    },
+                    _ => {
+                        return;
+                    }
+                };
+
+                self.visit_expr(right);
+                let right_expr_type = match self.result.take() {
+                    Some(Ty::Match(m)) => m.matchstr,
+                    Some(Ty::PathSearch(ps)) => match ps.resolve_as_match(self.session) {
+                        Some(m) => m.matchstr,
+                        _ => {
+                            return;
+                        }
+                    },
+                    _ => {
+                        return;
+                    }
+                };
+                let searchstr = match bin.node {
```

Comment:
I think these codes should go to `typeinf.rs`.
We shouldn't make a function too long.